### PR TITLE
DO NOT MERGE Ana/fix extension redirecting. add testnet checkbox

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Adds a critical testnet checkbox for extension signin

Thank you! 🚀

---

For contributor:

- [x] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by Bitcoinera** [comment]
Let's get this merged 💃 !

File: src/components/common/TmSessionExtension.vue [hunk 1]
```diff
@@ -68,20 +79,45 @@ export default {
...
         sessionType: `extension`,
         address: account.address,
-        networkId: account.network ? account.network : "cosmos-hub-mainnet" // defaulting to cosmos-hub-mainnet
+        networkId: network.id
       })
     },
     async signInAndRedirect(account) {
       await this.signIn(account)
+      const accountNetwork = this.getSignInNetwork(account)
+      console.log(accountNetwork)
```
Comment (jbibla) [inline src/components/common/TmSessionExtension.vue]: remove console.log

Comment (Bitcoinera) [reply src/components/common/TmSessionExtension.vue]: xD

File: src/components/common/TmSessionExtension.vue [hunk 2]
```diff
@@ -37,22 +37,33 @@
         </p>
       </div>
     </div>
+    <TmFormGroup class="field-checkbox" field-id="sign-up-warning" field-label>
+      <div class="field-checkbox-testnet">
+        <label class="field-checkbox-label" for="select-testnet">
+          <input id="select-testnet" v-model="testnet" type="checkbox" />
+          This is a testnet address</label
```
Comment (faboweb) [inline src/components/common/TmSessionExtension.vue]: @jbibla How about something like: Until we allow to copy addresses from testnet to mainnet you can decide the target network here.

File: src/components/common/TmSessionExtension.vue [hunk 3]
```diff
@@ -68,20 +79,45 @@ export default {
...
       this.$store.dispatch("getAddressesFromExtension")
     },
+    getSignInNetwork(account) {
+      // defaulting to cosmos-hub-mainnet
+      const accountNetwork = account.network || "cosmos-hub-mainnet"
+      if (this.testnet) {
+        return (
+          this.networks
+            .filter(({ testnet }) => testnet)
+            .find(({ slug }) => accountNetwork.startsWith(slug)) ||
```
Comment (faboweb) [inline src/components/common/TmSessionExtension.vue]: 🤮 but ok for now

File: src/components/common/TmSessionExtension.vue (hunk 1 above)
Comment (faboweb) [inline src/components/common/TmSessionExtension.vue]:
```suggestion
```

File: src/components/common/TmSessionExtension.vue (hunk 3 above)
Comment (Bitcoinera) [reply src/components/common/TmSessionExtension.vue]: 😿

Comment (Bitcoinera) [reply src/components/common/TmSessionExtension.vue]: I mean, I also tried `.find(({ slug, testnet }) => accountNetwork.startsWith(slug) && testnet)`, but that somehow didn't work 🤷‍♀

**Comment by faboweb** [comment]
Will stay open to provide preview to users in need

File: src/components/common/TmSessionExtension.vue (hunk 3 above)
Comment (jbibla) [reply src/components/common/TmSessionExtension.vue]: 🤮 is much worse than 🤢 fabo

🌈 is the lunie way 🌈

**Comment by Bitcoinera** [comment]
Oops, forgot about this one. What is the plan with this? Do we have the new designs?

**Comment by jbibla** [comment]
we can probably close it and re-open if someone needs it.

**Comment by Bitcoinera** [comment]
But right now it is not very ideal how the extension login is working... Why not just add this checkbox for now?

I think it is already better UX than what we have now. Even for us developers

**Comment by Bitcoinera** [comment]
Why don't we merge this? I mean, we have the exact same logic for insecure Sign in, and this Sign In we will need to "open" it anyways once we release Polkadot on production...

What is the plan here?

**Comment by jbibla** [comment]
i think this is very unique edge case and should not happen again in the future. the proper solution from my perspective is to allow seed recovery and add the mainnet / testnet tabs so it’s clearer.

**Comment by jbibla** [comment]
i think this is very unique edge case and should not happen again in the future. 

the proper solution from my perspective is to allow seed recovery #3873, one seed to rule them all #3782 and add the mainnet / testnet tabs #3872.

**Pull request closed without merging**